scripts/inference.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import os
+
+import pandas as pd
+import torch
+from datasets import load_dataset
+from tqdm.auto import tqdm
+from transformers import AutoModelForCausalLM
+
+torch.set_grad_enabled(False)
+
+
+def get_correct_logprobs(model, samples_tok):
+    # logits: seq, pos, d_vocab
+    logits = model(samples_tok).logits
+    # logprobs: [batch_size, seq_length, vocab_size]
+    logprobs = torch.nn.functional.log_softmax(logits, dim=-1)
+
+    # make probs a list of lists of correct token LOG probabilities.
+    list_logprob = []
+    for i, sample in enumerate(samples_tok):
```

Comment:
It should be quite straightforward to do it for all samples at once, w/o the for loop